xarray/test/test_dataarray.py

```diff
@@ -1407,6 +1407,32 @@ def test_resample_first(self):
                              name='time')
         self.assertDataArrayIdentical(expected, actual)
 
+    def test_resample_first_keep_attrs(self):
+        times = pd.date_range('2000-01-01', freq='6H', periods=10)
+        array = DataArray(np.arange(10), [('time', times)])
+        array.attrs['meta'] = 'data'
```

Comment:
you may consider using the fixtures we create [here](https://github.com/mcgibbon/xarray/blob/master/xarray/test/test_dataarray.py#L17-L22).  You could get away with:

```Python
array = self.dv
array.attrs = self.attrs

# later on...
assert array.attrs == self.attrs
```